utils/swift_build_support/swift_build_support/products/earlyswiftdriver.py

```diff
@@ -0,0 +1,133 @@
+# swift_build_support/products/swiftdriver.py -------------------*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+
+import os
+
+from . import product
+from .. import shell
+from .. import toolchain
+
+
+# A "Special" SwiftDriver that gets build with the host toolchain to act
+# as *the* compiler driver for the `swift` build directory compiler,
+# hence it does not depend on any other build product of `build-script`.
```

Comment:
I would add more documentation here around the special behavior of this specifically:

1. I would add that we don't build if we don't find in the host toolchain. But we may error in the future.
2. I would add that the application is written in Swift.
....

The big thing for me is that I want someone who is reading earlyswiftdriver.py/swiftdriver.py to be able to understand the difference in between them without searching through the rest of the repository.

Reply:
Added a longer explanation here, and a comment to `swiftdriver.py`.